server/api/api/endpoints/nuclio.py

```diff
@@ -527,6 +527,8 @@ def _deploy_nuclio_runtime(
                         "On deploy of serving-functions which is based on mlrun image "
                         "('mlrun/') and with set-tracking enabled, client version must be >= 1.6.3"
                     )
+
+                # Update model monitoring stream trigger if using old V3IO stream path
```

Comment:
i think some refactoring can add to readblity here.
* is_ce_mode() is also checking along these lines
* the check in is_monitoring_stream_has_the_new_stream_trigger() is really checking if the monitoring needs to be redeployed (ex for kafka it does not)
maybe something like
if serving_to_monitor:
   if monitoring_deployment.is_monitoring_stream_redeploy_needed():
        if (fn.spec.image.startswith("mlrun/")...):
            ...    
            raise mlrun.errors.MLRunBadRequestError(..)
                
        monitoring_deployment.deploy_model_monitoring_stream_processing(
                        overwrite=True
                    )